base/src/freetrm/eventstore/http/EventStoreHttpServer.scala

```diff
@@ -0,0 +1,254 @@
+              user =>
+                pathPrefix("consume") {
+                  path(Segment / LongNumber / LongNumber.?) {
+                    (topic, earliestOffsetToReadFrom, lastOffsetToReadTo) =>
+                      complete {
+                        consume(topic, earliestOffsetToReadFrom, lastOffsetToReadTo)
+                      }
+                  }
+                } ~
+                pathPrefix("consumeNotifications") {
+                  complete {
+                    consumeNotifications
+                  }
+                } ~
+                pathPrefix("topicExists") {
+                  path(Segment) {
+                    (topic) =>
+                      complete {
+                        topicExists(topic)
+                      }
+                  }
+                } ~
+                pathPrefix("surface") {
+                  path(Segment) {
+                    (topic) =>
+                      complete {
+                        surface(topic)
+                      }
+                  }
+                } ~
+                pathPrefix("maxTxnNo") {
+                  path(Segment) {
+                    (topic) =>
+                      complete {
+                        maxTxnNo(topic)
+                      }
+                  }
+                } ~
+                pathPrefix("maxSeqNo") {
+                  path(Segment) {
+                    (topic) =>
+                      complete {
+                        maxSeqNo(topic)
+                      }
+                  }
+                } ~
+                pathPrefix("listTopics") {
+                  complete {
+                    listTopics
+                  }
+                }
+            }
+          }
+        }
+      }
+  }
+
+  def writeToEventStore(topic: String, event: Event): Future[HttpResponse] = {
+    execAndHandle(
+      () => dedup.writeWithDuplicateCheckResult(topic, event).map {
+        case result if !result.wasDuplicate =>
+          response(OK, EventStoreResponseOK(result.version))
+        case result =>
+          response(OK, EventStoreResponseDuplicate(result.version))
+      },
+      e =>
+        s"Failed to publish to event store: $topic: $event"
+    )
+  }
+
+
+  def consume(topic: String,
+              earliestOffsetToReadFrom: Long,
+              lastOffsetToReadTo: Option[Long]): Source[ServerSentEvent, NotUsed] = {
+
+    reader
+      .consume(topic, earliestOffsetToReadFrom, lastOffsetToReadTo)
+      .map {
+        e => ServerSentEvent(e.toJson.prettyPrint)
+      }
+      .keepAlive(1.second, () => {
+        ServerSentEvent.heartbeat
+      })
+  }
+  
+  def consumeNotifications: Source[ServerSentEvent, NotUsed] = {
+    reader
+      .consumeNotifications
+      .map {
+        e => ServerSentEvent(e.toJson.prettyPrint)
+      }
+      .keepAlive(1.second, () => {
+        ServerSentEvent.heartbeat
+      })
+  }
+
+  def surface(topic: String): Source[ServerSentEvent, NotUsed] = {
```

Comment:
`newestSurface` perhaps? `surface` is used elsewhere in the system to mean the latest as of a particular version